### PR TITLE
AAC-546 - Add ASN and NI to defendants search

### PR DESCRIPTION
**Pull request by pking-moj**
## Description of change

We need the additional search options of NI and ASN on the defendants endpoint to provide feature parity. This adds this in with testing.

## Link to Jira Ticket

- [AAC-546](https://dsdmoj.atlassian.net/browse/AAC-546)

## Notes

Test data is hard to come by, im not sure if this is working from a CDA/CP point of view. However it is implement as specified

### CODE REVIEW

**Comment by ivanELEC** [comment]
Note that we have a few examples in Last Pass of Dev ASNs and NINOs - I used these to test

File: laa_court_data_api_app/routers/defendants.py [hunk 1]
```diff
@@ -33,6 +34,14 @@ async def get_defendants(urn: Optional[str] = None,
...
         cda_response = await client.get("/api/internal/v2/prosecution_cases",
                                         params={"filter[prosecution_case_reference]": urn})
+    elif asn:
+        logging.info(f"Defendants_Get_Urn_{asn}")
+        cda_response = await client.get("/api/internal/v2/prosecution_cases",
+                                        params={"filter[arrest_summons_number]": asn})
+    elif nino:
+        logging.info(f"Defendants_Get_Urn_{nino}")
+        cda_response = await client.get("/api/internal/v2/prosecution_cases",
+                                        params={"filter[national_insurance_number]": nino})
```
Comment (ivanELEC) [inline laa_court_data_api_app/routers/defendants.py:44]: Logs are a little bit misleading as they all indicate Get_Urn even though we are getting via ASN

File: postman/collections/defendants.postman_collection.json [hunk 2]
```diff
@@ -225,6 +225,116 @@
...
+					"path": [
+						"v2",
+						"defendants"
+					],
+					"query": [
+						{
+							"key": "urn",
+							"value": "{{Nino}}"
+						}
+					]
```
[the remark this answers is not in the document]
Comment (pking-moj) [reply postman/collections/defendants.postman_collection.json]: Fixed

**Review by ivanELEC — APPROVED** [review]

**Review by Obsiye — APPROVED** [review]

File: laa_court_data_api_app/routers/defendants.py [hunk 3]
```diff
@@ -33,6 +34,14 @@ async def get_defendants(urn: Optional[str] = None,
         logging.info(f"Defendants_Get_Urn_{urn}")
         cda_response = await client.get("/api/internal/v2/prosecution_cases",
                                         params={"filter[prosecution_case_reference]": urn})
+    elif asn:
+        logging.info(f"Defendants_Get_asn_{asn}")
```
Comment (mabely) [inline laa_court_data_api_app/routers/defendants.py:38]: I'ma be really annoying and ask that 'asn' and 'nino' be pascal case just for predictability when searching through logs

**Review by mabely — CHANGES_REQUESTED** [review]

**Review by mabely — APPROVED** [review]

**Review by ivanELEC — APPROVED** [review]

**Review by Obsiye — APPROVED** [review]

**Pull request merged**